src/executor/plan_executor.py

```diff
@@ -98,27 +98,11 @@ def _clean_execution_tree(self, tree_root: AbstractExecutor):
         # ToDo
         # clear all the nodes from the execution tree
 
-    def execute_plan(self):
+    def execute_plan(self) -> Iterator[Batch]:
         """execute the plan tree
-
         """
-        # TODO: for now this returns list of batch frames. Update to return
-        # a stitched output
         execution_tree = self._build_execution_tree(self._plan)
-
-        output_batches = Batch(pd.DataFrame())
-
-        # ToDo generalize this logic
-        _INSERT_CREATE_LOAD = (
-            PlanNodeType.CREATE,
-            PlanNodeType.INSERT,
-            PlanNodeType.CREATE_UDF,
-            PlanNodeType.LOAD_DATA)
-        if execution_tree.node.node_type in _INSERT_CREATE_LOAD:
-            execution_tree.exec()
-        else:
-            for batch in execution_tree.exec():
-                output_batches += batch
-
+        output = execution_tree.exec()
+        if output is not None:
```

Comment:
We can use `yield from []` in the executors. And `yield from execution_tree.exec()` here to avoid this `output is not None` check.

Reply:
The reason for this logic is that I thought we shouldn't force all the executors to always yield. There are cases where you want to return instead of yielding or maybe someone implementing a new executor and missed yielding in one of the conditional branches (as there can be multiple exit points in the code). How do you feel about this?